futures-cpupool/src/lib.rs

```diff
@@ -252,3 +268,61 @@ impl<F: Future> Future for Sender<F, Result<F::Item, F::Error>> {
         Ok(Async::Ready(()))
     }
 }
+
+impl Builder {
+    /// Create a builder a number of workers equal to the number
+    /// of CPUs on the host.
+    pub fn new() -> Builder {
+        Builder {
+            pool_size: num_cpus::get(),
+            after_start: None,
+            before_stop: None,
+        }
+    }
+
+    /// Set size of a future CpuPool
+    pub fn pool_size(mut self, size: usize) -> Self {
+        self.pool_size = size;
+        self
+    }
+
+    /// Execute function `fun` before each worker thread
```

Comment:
Could this clarify that "before" is "right after the thread is spawned but before it does work"?

Also, typically closure variables are called things like `f` or `g` rather than longer names like "fun" I've seen at least